### PR TITLE
feat: Added utility function solveCubic()

### PR DESCRIPTION
**Pull request by st-pasha**
# Description

The function solves a cubic equation, and can be used for a variety of purposes, such as handling the Bezier curves.

## Checklist

- [x] The title of my PR starts with a [Conventional Commit] prefix (`fix:`, `feat:`, `docs:` etc).
- [x] I have read the [Contributor Guide] and followed the process outlined for submitting PRs.
- [x] I have updated/added tests for ALL new/updated/fixed functionality.
- [x] I have updated/added relevant documentation in `docs` and added dartdoc comments with `///`.
- [-] I have updated/added relevant examples in `examples`.

## Breaking Change

- [-] Yes, this is a breaking change.
- [x] No, this is *not* a breaking change.

## Related Issues

[issue database]: https://github.com/flame-engine/flame/issues
[Contributor Guide]: https://github.com/flame-engine/flame/blob/main/CONTRIBUTING.md
[Flame Style Guide]: https://github.com/flame-engine/flame/blob/main/STYLEGUIDE.md
[Conventional Commit]: https://conventionalcommits.org

### CODE REVIEW

**Review by spydon — APPROVED** [review]

File: packages/flame/lib/src/utils/solve_cubic.dart [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+/// Solves cubic equation `ax³ + bx² + cx + d == 0`.
+///
+/// Depending on the coefficients, either 1 or 3 real solutions may be returned.
+/// In degenerate cases, when some of the roots of the equation coincide, we
+/// return all such roots without deduplication.
+///
+/// If coefficient [a] is equal to zero, then we solve the equation as a
+/// quadratic one (see [solveQuadratic]).
+List<double> solveCubic(double a, double b, double c, double d) {
```
Comment (wolfenrain) [inline packages/flame/lib/src/utils/solve_cubic.dart:13]: I wonder if this should be in utils or if we should have a dedicated math section?

**Review by wolfenrain — APPROVED** [review]
LGTM, left a small comment that I kinda forgot in the previous PR😅

File: packages/flame/lib/src/utils/solve_cubic.dart (hunk 1 above)
Comment (st-pasha) [reply packages/flame/lib/src/utils/solve_cubic.dart:13]: "Math sees itself as a dreamy poet. Science sees it as a supplier of specialized technical equipment. And herein we find one of the greatest paradoxes of human inquiry: These two views, both valid, are hard to reconcile. If math is an equipment supplier, why is its equipment so strangely poetic? And if math is a poet, then why is its poetry so unexpectedly useful?"

Comment (spydon) [reply packages/flame/lib/src/utils/solve_cubic.dart:13]: Haha, that can be the top level doc for the math section :smile:
I agree that we can have a math directory and a barrel file for it, I'm guessing we'll have quite a lot of more things going in there later.

Comment (spydon) [reply packages/flame/lib/src/utils/solve_cubic.dart:13]: That can be done on a separate PR though since it will move the other file too

**Pull request merged**